src/base/profile_p.cpp

```diff
@@ -116,33 +126,38 @@ QString Private::DefaultProfile::locationWithConfigurationName(const QStandardPa
 
 Private::CustomProfile::CustomProfile(const QString &rootPath, const QString &configurationName)
     : Profile {configurationName}
-    , m_rootDirectory {QDir(rootPath).absoluteFilePath(this->profileName())}
+    , m_rootPath {QDir(rootPath).absolutePath()}
+{
+}
+
+QString Private::CustomProfile::rootPath() const
 {
+    return m_rootPath;
 }
 
-QString Private::CustomProfile::baseDirectory() const
+QString Private::CustomProfile::basePath() const
 {
-    return m_rootDirectory.canonicalPath();
+    return QDir(m_rootPath).absoluteFilePath(profileName());
 }
 
 QString Private::CustomProfile::cacheLocation() const
 {
-    return m_rootDirectory.absoluteFilePath(QLatin1String(cacheDirName));
+    return QDir(basePath()).absoluteFilePath(QLatin1String(cacheDirName));
```

Comment:
I think there are excessive conversions from/to `QDir`.
It should be possible to provide an internal function to return `QDir(m_rootPath)` directly.

Reply:
My initial intention was to initialize all paths like cacheLocation in constructor. How about it?

Reply:
>My initial intention was to initialize all paths like cacheLocation in constructor. How about it?

OK since they do not intended to change at runtime.

Reply:
>>My initial intention was to initialize all paths like cacheLocation in constructor. How about it?
>
>OK since they do not intended to change at runtime.

Done.